File: assign11/q1.cpp
#include <iostream>
#include <bits/stdc++.h>
using namespace std;

int squareroot(int n){
	int l = 0,r = n-1;
	while(l<r){
		int mid =floor((l+r)/2);
		if(mid*mid > n){
			r=mid-1;
		}
		else if(mid*mid < n){
			l=mid+1;
		}
		else{
			return mid;
		}
	}
	return l;
}

int main(){
	
	cout<<squareroot(4)<<endl;
	cout<<squareroot(8)<<endl;
	
	return 0;
}

File: surprise2/q1.py
# -*- coding: utf-8 -*-
"""
Created on Sun Jun 18 14:47:55 2023

@author: harsha
"""
import math

def squareroot(n):
    l = 0;r = n;
    while(l<r):
        mid = math.floor((l+r)/2)
        if(mid*mid == n):
            return mid
        elif mid*mid > n:
            r = mid
        else:
            l = mid+1
    return l-1
    

print(squareroot(64))
print(squareroot(8))
print(squareroot(4))